appliance/mariadb/handler.go

```diff
@@ -0,0 +1,133 @@
+package mariadb
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+	"io"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/sirenia/client"
+	"github.com/flynn/flynn/pkg/sirenia/state"
+	"github.com/flynn/flynn/pkg/status"
+)
+
+// Handler represents an HTTP API handler for the process.
+type Handler struct {
+	router *httprouter.Router
+
+	Process     *Process
+	Peer        *state.Peer
+	Heartbeater discoverd.Heartbeater
+	Logger      log15.Logger
+}
+
+// NewHandler returns a new instance of Handler.
+func NewHandler() *Handler {
+	h := &Handler{
+		router: httprouter.New(),
+		Logger: log15.New(),
+	}
+	h.router.Handler("GET", status.Path, status.Handler(h.healthStatus))
+	h.router.GET("/status", h.handleGetStatus)
+	h.router.GET("/backup", h.handleGetBackup)
+	h.router.POST("/stop", h.handlePostStop)
+	return h
+}
+
+// ServeHTTP serves an HTTP request.
+func (h *Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) { h.router.ServeHTTP(w, req) }
+
+// healthStatus returns the current status of the peer and process.
+func (h *Handler) healthStatus() status.Status {
+	info := h.Peer.Info()
+	if info.State == nil || info.RetryPending != nil ||
+		(info.Role != state.RolePrimary && info.Role != state.RoleSync && info.Role != state.RoleAsync) {
+		return status.Unhealthy
```

Comment:
Yes. Is it possible to extract and re-use these routes and compose them with the extra MariaDB ones?